app/helpers/reports_helper.rb

```diff
@@ -41,6 +43,88 @@ def render_report_element(element, provided_locals = nil)
+
+        locals[:filename] = obj_name
+        locals[:path] = "#{obj_folder_name}/#{obj_name}"
+      elsif element['type_of'].in? %w(step_asset step_table result_asset
+                                      result_table)
+
+        parent_el = ReportElement.find(element['parent_id'])
+        parent_type = parent_el[:type_of]
+        parent = parent_type.singularize.classify.constantize
+                            .find(parent_el["#{parent_type}_id"])
+
+        if parent.class == Step
+          obj_name = if element['type_of'] == 'step_asset'
+                       name = Asset.find(element[:asset_id]).file_file_name
+                       suffix = name.split('.').second
+                       suffix.prepend('.') if suffix
+                       name.split('.').first
+                     else
+                       name = Table.find(element[:table_id]).name
+                       suffix = '.csv'
+                       name.empty? ? 'Table' : name
+                     end
+          obj_name = to_filesystems_compatible_filename(
+            obj_name.truncate(
+              Constants::EXPORTED_FILE_NAME_TRUNCATION_LENGTH,
+              omission: ''
+            )
+          )
+          obj_name += "_Step#{parent.position + 1}#{suffix}"
+          obj_folder_name = 'Protocol attachments'
+          parent_module = if parent.protocol.present?
+                            parent.protocol.my_module
+                          else
+                            parent.my_module
+                          end
+        else
+          obj_name = if element['type_of'] == 'result_asset'
+                       name = Asset.find(element[:result_id]).file_file_name
+                       suffix = name.split('.').second
+                       suffix.prepend('.') if suffix
+                       name.split('.').first
+                     else
+                       name = Result.find(element[:result_id]).name
+                       suffix = '.csv'
+                       name.empty? ? 'Table' : name
+                     end
+          obj_name = to_filesystems_compatible_filename(
+            obj_name.truncate(
+              Constants::EXPORTED_FILE_NAME_TRUNCATION_LENGTH,
+              omission: ''
+            )
+          )
+          obj_name += suffix
+          obj_folder_name = 'Results attachments'
+          parent_module = parent
+        end
+
+        parent_module_name = to_filesystems_compatible_filename(
+          parent_module.name.truncate(
+            Constants::EXPORTED_FILE_NAME_TRUNCATION_LENGTH,
+            omission: ''
+          )
+        )
+        parent_exp_name = to_filesystems_compatible_filename(
+          parent_module.experiment.name.truncate(
+            Constants::EXPORTED_FILE_NAME_TRUNCATION_LENGTH,
+            omission: ''
+          )
+        )
+
+        locals[:filename] = obj_name
+        locals[:path] = "#{parent_exp_name}/#{parent_module_name}/" \
```

Comment:
Does not take into sequence number for all types as mentioned on the PR comment.